src/Components/PostForm.jsx

```diff
@@ -0,0 +1,105 @@
+import {
+  ref as storageRef,
+  uploadBytes,
+  getDownloadURL,
+} from "firebase/storage";
+import { push, ref, set, update } from "firebase/database"; 
+import { database, storage } from "../firebase";
+import { useState, useEffect } from "react";
+import { useNavigate } from 'react-router-dom';
+
+const DB_STORAGE_KEY = "images"; 
+const DB_MESSAGES_KEY = "messages";
+
+function PostForm({isLoggedIn, user, editing, setEditing, setEditingData, editingData}) {
+  const [textInput, setTextInput] = useState("");
+  const [fileInputFile, setFileInputFile] = useState(null);
+  const [currentURL, setCurrentURL] = useState("");
+  const [currentLikes, setCurrentLikes] = useState(0);
+  const [currentDislikes, setCurrentDislikes] = useState(0);
+
+  const messagesRef = ref(database, DB_MESSAGES_KEY);
+  const navigate = useNavigate();
+
+  const writeData = (e) => {
+    e.preventDefault();
+    const newMessageRef = push(messagesRef);
+    const storageRefInstance = storageRef(
+      storage,
+      DB_STORAGE_KEY + "/" + fileInputFile.name
+    );
+    try {
```

Comment:
I would try to combine likes and dislikes, instead of having to calculate both when rendering out, you can just showcase a positive or negative number - 
When updating likes look at the above comment when storing the data (can store UID, so users can only like once)